GeneratorInterface/Pythia8Interface/src/CustomHookFactory.cc

```diff
@@ -0,0 +1,3 @@
+#include "GeneratorInterface/Pythia8Interface/interface/CustomHook.h"
+
+EDM_REGISTER_PLUGINFACTORY(CustomHookFactory, "CustomHookFactory");
```

Comment:
it could also be useful to define a shorthand macro for registering userhooks, like `REGISTER_USERHOOK(SuepDecay)` (example: https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/SimGeneral/MixingModule/interface/DigiAccumulatorMixModFactory.h#L31)